uid2_client/identity_map_input.py

```diff
@@ -0,0 +1,62 @@
+import json
+
+from uid2_client import IdentityType, normalize_and_hash_email, normalize_and_hash_phone
+
+
+class IdentityMapInput:
+    """input for IdentityMapClient, such as email addresses or phone numbers"""
+
+    def __init__(self, identity_type, emails_or_phones, already_hashed):
+        self.hashed_dii_to_raw_diis = {}
+        self.hashed_normalized_emails = None
+        self.hashed_normalized_phones = None
+        if identity_type == IdentityType.Email:
+            if already_hashed:
+                self.hashed_normalized_emails = emails_or_phones
+            else:
+                self.hashed_normalized_emails = []
+                for email in emails_or_phones:
+                    hashed_normalized_email = normalize_and_hash_email(email)
+                    self._add_hashed_to_raw_dii_mapping(hashed_normalized_email, email)
+                    self.hashed_normalized_emails.append(hashed_normalized_email)
+        else:  # phone
+            if already_hashed:
```

Comment:
nit: `if already_hashed` block should not depend on whether it's an email or phone and should go outside the phone/email check

Reply:
Hmm, it depends on the type. If it's a hashed email list, it gets assigned to `hashed_normalized_emails`, and if it's a hashed phone list, it's assigned to `hashed_normalized_phones`.